src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx

```diff
@@ -0,0 +1,133 @@
+
+class PublishRequestFormModel extends FormModel {
+  modal: SentryAppPublishRequestModal;
+
+  constructor(modal, ...args) {
+    super(...args);
+    this.modal = modal;
+  }
+
+  getTransformedData() {
+    const data = this.getData();
+    //map object to list of questions
+    const questionnaire = this.modal.formFields.map(field => {
+      return {
+        question: field.label,
+        answer: data[field.name],
+      };
+    });
+    return {questionnaire};
+  }
+}
+
+type Props = {
+  app: SentryApp;
+  location: Location;
+  closeModal: () => void;
+};
+
+export default class SentryAppPublishRequestModal extends React.Component<Props> {
+  static propTypes = {
+    app: PropTypes.object.isRequired,
+  };
+
+  form = new PublishRequestFormModel(this);
+
+  get formFields() {
+    const {app} = this.props;
+    //replace the : with a . so we can reserve the colon for the question
+    const scopeString = app.scopes.join(', ').replace(/:/g, '-');
+    const baseFields = [
+      {
+        type: 'textarea',
+        required: true,
+        label: 'What does your integration do? Please be as detailed as possible.',
+        autosize: true,
+      },
+      {
+        type: 'textarea',
+        required: true,
+        label: 'What value does it offer customers?',
+        autosize: true,
+      },
+      {
+        type: 'textarea',
+        required: true,
+        label: `Please justify why you are requesting each of the following scopes: ${scopeString}.`,
+        autosize: true,
+      },
+      {
+        type: 'textarea',
+        required: true,
+        label: 'Do you operate the web service your integration communicates with?',
+        autosize: true,
+      },
+    ];
+    //dynamically generate the name based off the index
+    return baseFields.map((field, index) =>
+      Object.assign({name: `question${index}`}, field)
+    );
+  }
+
+  onSubmitSuccess = () => {
+    addSuccessMessage(t('Request to publish %s successful.', this.props.app.slug));
+    this.props.closeModal();
+  };
+
+  onSubmitError = () => {
+    addErrorMessage(t('Request to publish %s fails.', this.props.app.slug));
+  };
+
+  render() {
+    const {app} = this.props;
+    const endpoint = `/sentry-apps/${app.slug}/publish-request/`;
+    const forms = [
+      {
+        title: t('Questions to answer'),
+        fields: this.formFields,
+      },
+    ];
+    return (
+      <React.Fragment>
+        <Header>{t('Publish Request Questionnaire')}</Header>
+        <Body>
+          <Explanation>
+            {t(
+              `Please fill out this questionnaire in order to get your integration evaluated for publication.
+              Once your integration has been approved, users outside of your organization will be able to install it.`
+            )}
+          </Explanation>
+          <Form
+            allowUndo
+            apiMethod="POST"
+            apiEndpoint={endpoint}
+            onSubmitSuccess={this.onSubmitSuccess}
```

Comment:
Can we rename this to `this.handleSubmitSuccess`? (https://github.com/getsentry/frontend-handbook#event-handlers)